src/set.rs

```diff
@@ -107,5 +146,385 @@ where
     }
 }
 
-// TODO: Tests
-// TODO: Iterators
+// TODO: impl Debug for ConSet<T, S>
+
+impl<T, S> ConSet<T, S>
+where
+    T: Clone + Hash + Eq + 'static,
+{
+    /// Returns an iterator through the elements of the set.
+    pub fn iter(&self) -> Iter<T, S> {
+        Iter {
+            inner: raw::iterator::Iter::new(&self.raw),
+        }
+    }
+}
+
+/// The iterator of the [`ConSet`].
+///
+/// See the [`iter`][ConSet::iter] method for details.
+pub struct Iter<'a, T, S>
+where
+    T: Clone + Hash + Eq + 'static,
+{
+    inner: raw::iterator::Iter<'a, TrivialConfig<T>, S>,
+}
+
+impl<'a, T, S> Iterator for Iter<'a, T, S>
+where
+    T: Clone + Hash + Eq + 'static,
+{
+    type Item = T;
+
+    fn next(&mut self) -> Option<T> {
+        match self.inner.next() {
+            Some(t) => Some(t.clone()),
+            None => None,
+        }
```

Comment:
nitpick: Wouldn't `self.inner.next().cloned()` be shorter/more readable?

Reply:
I love _"nitpicking"_! And I completely agree.